app/models/miq_schedule.rb

```diff
@@ -217,6 +217,18 @@ def action_check_compliance(obj, _at)
     _log.info("Action [#{name}] has been run for target type: [#{obj.class}] with name: [#{obj.name}]")
   end
 
+  def action_automation_request(_klass, _at)
+    uri_parts = {}.tap do |part|
+      part['class'] = filter[:uri_parts][:class] || "Process"
+      part['namespace'] = filter[:uri_parts][:namespace] || "/System"
+      part['instance'] = filter[:uri_parts][:instance]
+      part['message'] = filter[:uri_parts][:message]
+    end
+    parameters = filter[:parameters]
+    user = User.find_by_userid(userid) || User.find_by_userid("admin")
+    AutomationRequest.create_from_scheduled_task(user, uri_parts, parameters)
```

Comment:
Maybe something like this?

```ruby
def action_automation_request(_klass, _at)
  parameters = filter[:parameters]
  user = User.find_by_userid(userid) || User.find_by_userid("admin")
  AutomationRequest.create_from_scheduled_task(user, filter[:uri_parts], parameters)
end
```

Let AutomationRequest figure out how to construct the uri parts.